app/models/allocation_history.rb

```diff
@@ -0,0 +1,13 @@
+class AllocationHistory < ApplicationStruct
+  attribute :user_id, Types::Uuid
+
+  def items
+    @items ||= events.map { |event| AllocationHistoryItem.new(event) }
+  end
+
+  private
+
+  def events
+    Allocating.user_events(user_id).backward
```

Comment:
"backwards" is already applied to this scope. I wonder if it should be removed from "Allocating.user_events" and added by the caller where needed? I.e. Allocating.user_competencies and this method.

Reply:
Ah good spot! Removed

Reply:
Ah good spot! That seems best to me to keep `user_events` flexible. I'll make that change